packages/xen-api/transports/json-rpc.mjs

```diff
@@ -1,26 +1,29 @@
 import { format, parse } from 'json-rpc-protocol'
+import { request } from 'undici'
 
 import XapiError from '../_XapiError.mjs'
 
 import UnsupportedTransport from './_UnsupportedTransport.mjs'
 
 // https://github.com/xenserver/xenadmin/blob/0df39a9d83cd82713f32d24704852a0fd57b8a64/XenModel/XenAPI/Session.cs#L403-L433
-export default ({ agent, client, url }) => {
+export default ({ dispatcher, url }) => {
   url = new URL('./jsonrpc', Object.assign(new URL('http://localhost'), url))
-  const path = url.pathname + url.search
 
   return async function (method, args) {
-    const res = await client.request({
+    const res = await request(url, {
+      dispatcher,
       body: format.request(0, method, args),
       headers: {
         Accept: 'application/json',
         'Content-Type': 'application/json',
       },
       method: 'POST',
-      path,
-      agent,
     })
 
+    if ((res.statusCode / 100) | (0 !== 2)) {
+      throw new Error('unexpect statusCode ' + res.statusCode)
+    }
+
```

Comment:
This is just a detail, but shouldn't we check the `content-type` before the `statusCode` ? If we have both incorrect `content-type` and bad `statusCode`, I feel like it makes more sense to throw the error related to the transport.

Reply:
I don't think this matters much, but even so I would say no, the status code is expected to be 200 even if it's another transport.